### PR TITLE
[Spring] Add documentation for spring object factory

### PR DESCRIPTION
**Pull request by mlvandijk**
## Summary

Add some documentation for cucumber-spring.

## Details

Adds some documentation for cucumber-spring; i.e. how to get started.

## Motivation and Context

Currently no documentation at all.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [ ] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by mlvandijk** [comment]
Some old work I had started for #724

File: spring/README.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+</dependencies>
+```
+
+## Annotation Based Configuration
+
+For your own classes:
+
+* Add the `@Component` annotation to each of the classes cucumber-spring should manage.
+
+* Add a `@Scope("cucumber-glue")` annotation to have cucumber-spring remove them **after each scenario**
```
Comment (mpkorstanje) [inline spring/README.md]: This strikes me as counter intuitive but it is how the code is working.

Comment (mlvandijk) [reply spring/README.md]: Does this need still to be updated? I'm open to improvements, but figured *some* docs are better than *no* docs ;) (unless blatantly incorrect)

Comment (mpkorstanje) [reply spring/README.md]: That is why merged already. If you want you can add it.

**Pull request merged**